pyscf/adc/radc.py

```diff
@@ -529,12 +554,20 @@ def kernel_gs(self):
         mem_incore = (max(nao_pair**2, nmo**4) + nmo_pair**2) * 8/1e6
         mem_now = lib.current_memory()[0]
 
-        if (self._scf._eri is None or
-            (mem_incore+mem_now >= self.max_memory and not self.incore_complete)):
+        if getattr(self, 'with_df', None) or getattr(self._scf, 'with_df', None):  
+           if getattr(self, 'with_df', None): 
+               self.with_df = self.with_df
+           else :
+               self.with_df = self._scf.with_df
 
-            def outcore_transform():
-                return radc_ao2mo.transform_integrals_outcore(self)
-            self.transform_integrals = outcore_transform
+           def df_transform():
+               return radc_ao2mo.transform_integrals_df(self)
+           self.transform_integrals = df_transform
+        elif (self._scf._eri is None or
+            (mem_incore+mem_now >= self.max_memory and not self.incore_complete)):
+           def outcore_transform():
+               return radc_ao2mo.transform_integrals_outcore(self)
+           self.transform_integrals = outcore_transform
```

Comment:
When DF is used in kernel_gs, will this guarantee it is being used in excited-state kernel and that we do not need to recompute integrals?